src/Dofs/sparsity_pattern.jl

```diff
@@ -96,21 +100,78 @@ function _coupling_to_local_dof_coupling(dh::DofHandler, coupling::AbstractMatri
     return outs
 end
 
-function _create_sparsity_pattern(dh::AbstractDofHandler, ch#=::Union{ConstraintHandler, Nothing}=#, sym::Bool, keep_constrained::Bool, coupling::Union{AbstractMatrix{Bool},Nothing})
+"""
+    cross_element_coupling!(dh::DofHandler, topology::ExclusiveTopology, sym::Bool, keep_constrained::Bool, couplings::Union{AbstractVector{<:AbstractMatrix{Bool}},Nothing}, cnt::Int, I::Vector{Int}, J::Vector{Int})
+
+Mutates `I, J` to account for cross-element coupling in fields with discontinuous interpolations.
+Returns the updated value of `cnt`.
+
+Used internally for sparsity patterns with discontinuous interpolations.
+"""
+function cross_element_coupling!(dh::DofHandler, topology::ExclusiveTopology, sym::Bool, keep_constrained::Bool, couplings::Union{AbstractVector{<:AbstractMatrix{Bool}},Nothing}, cnt::Int, I::Vector{Int}, J::Vector{Int})
+    for (sdh_idx, sdh) in pairs(dh.subdofhandlers)
+        # Buffering interpolation information for type stability/better allocations
+        ip_infos = InterpolationInfo[]
+        nbasefunctions = Int[]
+        for ip in sdh.field_interpolations
+            ip_info = InterpolationInfo(ip)
+            push!(ip_infos, ip_info)
+            push!(nbasefunctions, getnbasefunctions(ip))
+        end
+        isnothing(couplings) || (coupling_sdh = couplings[sdh_idx])
+        for (cell_field_i, cell_field) in enumerate(sdh.field_names)
+            fii = ip_infos[cell_field_i]
+            # Couple element for discontinuous interpolations only
+            fii.is_discontinuous || continue
+            for cell_idx in eachindex(getcells(dh.grid))
+                # Making sure current field is defined for current cell
+                cell_field ∈ dh.subdofhandlers[dh.cell_to_subdofhandler[cell_idx]].field_names || continue
+                dofrange1 = dof_range(sdh, cell_field)
+                cell_dofs = celldofsview(dh, cell_idx)
+                cell_field_dofs = @view cell_dofs[dofrange1]
+                # For 1D case, cells must share faces to be neighbors. Otherwise use cell_face_neighbor for higher dimensions
+                for neighbor_cell in (getdim(dh.grid.cells[cell_idx]) >1 ? topology.cell_face_neighbor[cell_idx] : topology.cell_neighbor[cell_idx])
+                    for (neighbor_field_i, neighbor_field) in enumerate(sdh.field_names)
+                        fii2 = ip_infos[neighbor_field_i]
+                        sdh2 = dh.subdofhandlers[dh.cell_to_subdofhandler[neighbor_cell.idx]]
+                        neighbor_field ∈ sdh2.field_names && fii2.is_discontinuous || continue
```

Comment:
So just to be clear, coupling is added only if the neighbor interpolation is also discontinuous? I think maybe we even warn in DofHandler construction now if all interpolations for a field don't belong to the same "family".

Reply:
You mean I should implement this : https://github.com/Ferrite-FEM/Ferrite.jl/blob/ccd0e71570993ee91c3adaa4ab97569f21a476e1/src/Dofs/DofHandler.jl#L231 ? If so, check for same interpolation or check for same family?

Reply:
No you don't have to do that, not in this PR. I just meant that if we add that check then this should be fine, because then `:u` must always be discontinuous

Reply:
(I think the interpolation type (Lagrange, DiscontinuousLagrange, ...), and the order, should be checked, but refshape might differ).